### ISSUE TITLE
Are categorical plots with single letter strings limited to show 10 categories?

### ISSUE THREAD

**Issue by ImportanceOfBeingErnest** [matplotlib/matplotlib#9843]
It seems categorical plots are limited to show 10 categories correctly when single letters are used as categories. The eleventh category is shown at the place of the second one. 

```
import matplotlib.pyplot as plt

x = list("ABCDEFGHIJK")
y = range(len(x))
plt.plot(x,y, marker="o")

plt.show()
```
![image](https://user-images.githubusercontent.com/23121882/33190683-9bd27a4e-d0af-11e7-97c9-216e0cc4e1c0.png)

In case at least one of the categories has more letters, `x.append("WORD")`, the plot is shown correctly:

![image](https://user-images.githubusercontent.com/23121882/33190687-be6bc056-d0af-11e7-9de2-8259d17f0863.png)

(This issue was brouht up in [this Stackoverflow question](https://stackoverflow.com/questions/47464242/can-you-explain-why-the-plot-is-stopping-at-j-at-index-10))
[Produced with python 2.7, matplotlib 2.1, numpy 1.13.0rc2]

**Comment by anntzer** [comment]
I'd guess this is due to the current implementation of categorical mapping which casts indices to strings stored in the same array as the labels and back into numbers.  If the array has dtype S1 (for example), i.e. string of width 1, then the labels get truncated.
There are various PRs around that rework categorical support, they should hopefully fix this issue as well.

**Comment by story645** [comment]
I think you mean that the values get truncated, ie the 10 gets truncated to a 1 before it's cast back to an int. (The sequence and locs are stored correctly.)
```Python
mcat.StrCategoryConverter.convert(ax.xaxis.unit_data.seq, None, ax.xaxis)
```
yields
```
array([ 0.,  1.,  2.,  3.,  4.,  5.,  6.,  7.,  8.,  9.,  1.])
```

**Comment by tacaswell** [comment]
Closed by #9783
